src/server/auth/cmds/cmds_test.go

```diff
@@ -188,9 +191,82 @@ func TestModifyAdminsPropagateError(t *testing.T) {
 		`).Run())
 }
 
+func TestGetAndUseAuthToken(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration tests in short mode")
+	}
+	activateAuth(t)
+	defer deactivateAuth(t)
+
+	// Test both get-auth-token and use-auth-token; make sure that they work
+	// together with -q
+	require.NoError(t, tu.BashCmd("echo admin | pachctl auth login").Run())
+	require.NoError(t, tu.BashCmd(`
+	pachctl auth get-auth-token -q robot:marvin \
+	  | pachctl auth use-auth-token
+	pachctl auth whoami \
+	  | match 'robot:marvin'
+		`).Run())
+}
+
+func TestActivateAsRobotUser(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration tests in short mode")
+	}
+	// We need a custom 'activate' command, so reproduce 'activateAuth' minus the
+	// actual call
+	defer deactivateAuth(t) // unwind "activate" command before deactivating
+	activateMut.Lock()
+	defer activateMut.Unlock()
+	cmd := tu.Cmd("pachctl", "enterprise", "get-state")
+	out, err := cmd.Output()
+	require.NoError(t, err)
+	if string(out) != "ACTIVE" {
+		require.NoError(t, // Enterprise not active in the cluster. Activate it
+			tu.Cmd("pachctl", "enterprise", "activate", tu.GetTestEnterpriseCode()).Run())
+	}
+	// Logout (to clear any expired tokens) and activate Pachyderm auth
+	require.NoError(t, tu.BashCmd(`
+	pachctl auth logout
+	pachctl auth activate --initial-admin=robot:hal9000
+	pachctl auth whoami \
+		| match 'robot:hal9000'
+	`).Run())
+
+	// Make "admin" a cluster admins, so that deactivateAuth works
+	require.NoError(t,
+		tu.Cmd("pachctl", "auth", "modify-admins", "--add=admin").Run())
+}
+
+func TestActivateMismatchedUsernames(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration tests in short mode")
+	}
+	// We need a custom 'activate' command, so reproduce 'activateAuth' minus the
+	// actual call
+	activateMut.Lock()
```

Comment:
Seems like this could be made into a helper like `idempotentEnterpriseActivate()`

Reply:
Done